### ISSUE TITLE
Can we add a prop to add class to the generated `li.nav-item` element when using Tabs?

### ISSUE THREAD

**Issue by matronator** [react-bootstrap/react-bootstrap#6480]
### Is your feature request related to a problem? Please describe

I need to apply some classes not on the `.nav-link` element, but on its parent `.nav-item`, but there's currently no way to access it.

### Describe the solution you'd like

Add a new prop to add class to the nav-item element.

### Describe alternatives you've considered

There really are no alternatives. Maybe modifying the class after the element is generated by React, but that would go against everything React stands for.

### Additional context

¯\_(ツ)_/¯

**Comment by kyletsang** [comment]
Check the docs here: https://react-bootstrap.github.io/components/tabs/#custom-tab-layout

This provides more options for customization

**Comment by matronator** [comment]
Yeah, I read that, but that seems like too much work when I only need to change one class of one element, otherwise keep the layout exactly as is with simple `<Tabs>`.

**Comment by bhuvanaindukuri** [comment]
The rendering is similar to the first example under this link: https://getbootstrap.com/docs/5.0/components/navs-tabs/#javascript-behavior
@matronator - If my understanding is right, your requirement is to assign styling to li component instead of the button component.

**Comment by matronator** [comment]
> @matronator - If my understanding is right, your requirement is to assign styling to li component instead of the button component.

Exactly. I want to do some CSS styling, but since the `active` class gets added to the component one level deeper, I can't target the other elements next to it, because they're wrapped in the `li` which I'd like to add class to instead.

**Comment by bhuvanaindukuri** [comment]
@matronator - Is it possible to share a sample requirement/styling that you are currently unable to achieve? As @kyletsang mentioned customization is possible using the example provided. The components in this case are rendered as below:
![image](https://user-images.githubusercontent.com/29200083/200619436-7e8f9a83-2b9e-473b-8e80-01a57958ef63.png)

It is possible to customize the styling of nav-item as in below example:
.mb-3 .nav-item {
    background-color: aqua;
}

**Comment by matronator** [comment]
Okay, so I needed to add `.active` class to the `li` element, because I wanted to target the siblings of the `.active` element to give them special styling, as in:

```css
.nav-item.active ~ .nav-item > .nav-link {
    color: blue; /* or whatever */
}
```

<img width="717" alt="Snímek obrazovky 2022-11-08 v 20 24 50" src="https://user-images.githubusercontent.com/5470780/200659076-b11e1f96-4f84-4d72-ac41-5850ebfb5d61.png">

Because if you could only add `.active` to the `.nav-link` inside, then there would be no way of targeting the `.nav-link` inside the other `.nav-item` as there's no parent selector in CSS.

But in the end I found a workaround, by using the `:has()` selector, like this:

<img width="733" alt="Snímek obrazovky 2022-11-08 v 20 29 57" src="https://user-images.githubusercontent.com/5470780/200658897-c11e18f2-5446-4259-bde8-c25d553d410e.png">

But it's not ideal as the browser support for `:has` isn't great.

**Comment by bhuvanaindukuri** [comment]
Why can't this styling be applied at nav-link level? I tried the same and it seems ok to me. I tried adding the exact styling as below:

.nav-tabs .nav-item.show .nav-link, .nav-tabs .nav-link.active {
    color: #ff0000;
    background-color: #fff;
    border-color: #dee2e6 #dee2e6 #fff;
    transform: translateX(-100%);
}

As the documentation says, the idea of react-bootstrap is to stay in sync with bootstrap. In bootstrap it is applied at nav-link level. I don't think it is advisable to make this change unless we can think of a requirement which cannot be met.

**Comment by matronator** [comment]
> Why can't this styling be applied at nav-link level?

Because I posted just a part of the code, I need it to change based on the sibling of the active element, which only works on the top most element. Don't know how to describe it and I already changed my solution. I'll try to look for the SO answer I got it from and post a codepen for you to see.
